q2_fmt/_peds.py

```diff
@@ -466,11 +479,458 @@ def _create_masking(time_metadata, donor_df, recip_df, reference_column):
+            "Ref": ["donor2", "donor3", "donor1", "donor3",
+                    "donor1", "donor2"]}
+                                ).set_index('id')
+
+    recip_df = pd.DataFrame({
+        'id': ['sample1', 'sample2', 'sample3'],
+        'Feature1': [1, 0, 0],
+        'Feature2': [0, 1, 0],
+        'Feature3': [0, 0, 1]}).set_index('id')
+
+    _create_duplicated_recip_table(mismatched_df, recip_df)
+
+    pd.DataFrame({
+            'id': ["sample1", "sample1", "sample2", "sample2",
+                   "sample3", "sample3"],
+            'Feature1': [1, 1, 0, 0, 0, 0],
+            'Feature2': [0, 0, 1, 1, 0, 0],
+            'Feature3': [0, 0, 0, 0, 1, 1]}).set_index('id')
+    """
+    duplicated_table = \
+        mismatched_df.merge(recip_df, left_index=True, right_index=True,
+                            how='left').drop(mismatched_df.columns, axis=1)
+    return duplicated_table
+
+
+def _create_sim_masking(mismatched_df, donor_df, reference_column):
+    """Create a donor mask to mask recipient feature that aren't in the donor.
+
+    Creates a Donor Numpy array that duplicates donor samples to match up
+    with duplicated_table. This will mask recipient feature that aren't
+    in the donor.
+
+    Parameters
+    ----------
+    mismatched_df: pd.DataFrame
+        A Dataframe that contains recipient samples as the index and mismatched
+        donors as the values. A recipient sample will appear as many times as
+        there are mismatched donors to pair with.
+    donor_df: pd.DataFrame
+        A feature table of FMT donors.
+    reference_column: str
+        Name of the reference column in the Sample Metadata.
+
+    Returns
+    -------
+    donor_mask: ndarray
+        a numpy array of donor feature information with the same order as
+        duplicated_recip_table. This allows for easy np array math.
+
+    Examples
+    --------
+    mismatched_df = pd.DataFrame({
+            'id': ["sample1", "sample1", "sample2", "sample2",
+                   "sample3", "sample3"],
+            "Ref": ["donor2", "donor3", "donor1", "donor3",
+                    "donor1", "donor2"]}
+                                ).set_index('id')
+    donor_df = pd.DataFrame({
+            'id': ['donor1', 'donor2', 'donor3'],
+            'Feature1': [1, 0, 0],
+            'Feature2': [0, 1, 0],
+            'Feature3': [0, 0, 1]}).set_index('id')
+
+    _create_sim_masking(mismatched_df, donor_df, reference_column)
+
+    ndarray[[0, 1, 0],
+            [0, 0, 1],
+            [1, 0, 0],
+            [0, 0, 1],
+            [1, 0, 0],
+            [0, 1, 0]]
+    """
+    donors = mismatched_df[reference_column]
+    donor_index_masking = []
+    for donor in donors:
+        donor_index_masking.append(donor_df.index.get_loc(donor))
+    donor_df_np = donor_df.to_numpy()
+    donor_mask = donor_df_np[donor_index_masking]
+    return donor_mask
+
+
+def _simulate_uniform_distro(mismatched_peds, iterations):
```

Comment:
It might make sense to rename `iterations` as `k` in this function for clarity (since `k` is being passed to `random.choice`.